### PR TITLE
add ZeroLengthSectionAsEOF option to the v0 module

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

### CODE REVIEW

**Comment by mvdan** [comment]
FYI @aarshkshah1992 @dirkmc

File: util/util.go [hunk 1]
```diff
@@ -2,61 +2,36 @@ package util
...
 func ReadNode(br *bufio.Reader) (cid.Cid, []byte, error) {
 	data, err := LdRead(br)
 	if err != nil {
 		return cid.Cid{}, nil, err
 	}
+	// ReadCid used to panic or error on null padding.
+	// Instead, return a sentinel error to let the user decide what to do.
+	if len(data) == 0 {
+		return cid.Cid{}, nil, ErrZeroLengthSection
+	}
```
Comment (masih) [inline util/util.go:34]: I am curious why not take option here similar [to internal fork in v2](https://github.com/ipld/go-car/blob/master/v2/internal/carv1/util/util.go#L18)?

File: util/util.go [hunk 2]
```diff
@@ -2,61 +2,36 @@ package util
...
+
 func ReadNode(br *bufio.Reader) (cid.Cid, []byte, error) {
 	data, err := LdRead(br)
 	if err != nil {
 		return cid.Cid{}, nil, err
 	}
+	// ReadCid used to panic or error on null padding.
+	// Instead, return a sentinel error to let the user decide what to do.
+	if len(data) == 0 {
+		return cid.Cid{}, nil, ErrZeroLengthSection
```
Comment (willscott) [inline util/util.go:33]: cid.Undef ?

**Review by willscott — APPROVED** [review]

File: util/util.go (hunk 2 above)
Comment (mvdan) [reply util/util.go:33]: I'm just following the rest of the code :) I don't mind one way or another, but it seems that consistency is best, and the cleanup seems a bit out of place for a bugfix

**Pull request closed without merging**